project-base/tests/FrontendApiBundle/Functional/Advert/GetAdvertsTest.php

```diff
@@ -0,0 +1,232 @@
+use Shopsys\FrameworkBundle\Model\Advert\AdvertFacade;
+use Tests\FrontendApiBundle\Test\GraphQlTestCase;
+
+class GetAdvertsTest extends GraphQlTestCase
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Advert\AdvertDataFactory
+     * @inject
+     */
+    private $advertDataFactory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Advert\AdvertFacade
+     */
+    private $advertFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\FileUpload\FileUpload
+     * @inject
+     */
+    private $fileUpload;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Advert\Advert
+     */
+    private $advertWithImage;
+
+    /**
+     * @var \League\Flysystem\MountManager
+     * @inject
+     */
+    private $mountManager;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->advertFacade = $this->getContainer()->get(AdvertFacade::class);
+        $this->loadTestAdverts();
+    }
+
+    private function loadTestAdverts(): void
+    {
+        $domainId = $this->domain->getId();
+
+        $advert1 = $this->advertDataFactory->create();
+        $advert1->name = 'Test advert footer 1';
+        $advert1->type = Advert::TYPE_CODE;
+        $advert1->code = 'I <3 Shopsys';
+        $advert1->positionName = 'footer';
+        $advert1->hidden = false;
+        $advert1->domainId = $domainId;
+        $advert1->uuid = '19fe7c4b-ef93-4d33-aac3-f6eba0508f4d';
+        $this->advertFacade->create($advert1);
+
+        $advert2 = $this->advertDataFactory->create();
+        $advert2->name = 'Test advert header 1';
+        $advert2->type = Advert::TYPE_CODE;
+        $advert2->code = '<a href="/foo">Foo</a>';
+        $advert2->positionName = 'header';
+        $advert2->hidden = false;
+        $advert2->domainId = $domainId;
+        $advert2->uuid = 'e0b22920-d9ca-4270-a548-3e4fb8f212c9';
+        $this->advertFacade->create($advert2);
+
+        $testImageName = 'logo.png';
+        $localImagePath = 'local://' . __DIR__ . '/Resources/' . $testImageName;
+        $abstractImagePath = 'main://' . $this->fileUpload->getTemporaryDirectory() . '/' . $testImageName;
+        $this->mountManager->copy($localImagePath, $abstractImagePath);
+
+        $imageUploadData = new ImageUploadData();
+        $imageUploadData->uploadedFiles = [$testImageName];
+
+        $advert3 = $this->advertDataFactory->create();
+        $advert3->name = 'Test advert header 2';
+        $advert3->type = Advert::TYPE_IMAGE;
+        $advert3->positionName = 'header';
+        $advert3->hidden = false;
+        $advert3->domainId = $domainId;
+        $advert3->uuid = 'bb06fb2d-871a-4a54-8fa0-3bb995ac2650';
+        $advert3->image = $imageUploadData;
+        $advert3->link = 'https://shopsys.com';
+        $this->advertWithImage = $this->advertFacade->create($advert3);
+    }
+
+    public function tearDown(): void
+    {
+        $this->advertFacade->delete($this->advertWithImage->getId());
```

Comment:
Extra delete is for the uploaded image being deleted?

Reply:
Yes, exactly!